tutorials/identifying-first-day-no-snow/index.md

```diff
@@ -0,0 +1,516 @@
+that it is retained in the final mosaics. The function encodes:
+
+- Calendar DOY (`calDoy`): enumerated DOY from January 1st
+- Relative DOY (`relDoy`): enumerated DOY from user-defined `startDoy`
+- Milliseconds from Unix epoch (`millis`)
+- Year (`year`) - Note that the year is tied to the `startDoy` e.g., if
+`startDoy` is set at 183, the analysis will cross the new year, the year
+given to all pixels will be the first year, even if a particular image was
+collected on January 1 of the subsequent year.
+
+Note that two global variables are initialized (`startDate` and `startYear`)
+that will be redefined iteratively in a subsequent step.
+
+```js
+var startDate;
+var startYear;
+
+function addDateBands(img) {
+  // Get image date.
+  var date = img.date();
+  // Get calendar day-of-year.
+  var calDoy = date.getRelative('day', 'year');
+  // Get relative day-of-year; enumerate from user-defined startDoy.
+  var relDoy = date.difference(startDate, 'day');
+  // Get the date as milliseconds from Unix epoch.
+  var millis = date.millis();
+  // Add all of the above date info as bands to the snow fraction image.
+  var dateBands = ee.Image.constant([calDoy, relDoy, millis, startYear])
+    .rename(['calDoy', 'relDay', 'millis', 'year']);
+  return img.addBands(dateBands).toInt16().set('millis', millis);
+}
+```
+
+### 3\. Define an analysis mask
+
+Here is the opportunity to define a mask for your analysis. This mask can be
+used to constrain the analysis to certain latitudes
+(`ee.Image.pixelLonLat()`), land cover types, geometries, etc. In this case
+we will: 1) mask out water so that the analysis is confined to pixels over
+landforms only; 2) mask out pixels that have very few days of snow cover; 3)
+mask out pixels that are snow covered for a good deal of the year
+(e.g., glaciers).
+
+Import the MODIS water/land mask dataset, select the ‘water_mask’ band, and
+set all land pixels to value 1.
+
+```js
+var waterMask = ee.Image('MODIS/MOD44W/MOD44W_005_2000_02_24')
+  .select('water_mask')
+  .not();
+```
+
+Import the MODIS Snow Cover Daily Global 500m product and select the
+'NDSI_Snow_Cover' band.
+
+**Note that this dataset is used in following
+sections. If you decide not to do any analysis masking, you must still
+include and run this line before the start of the next section**.
+
+```js
+var completeCol = ee.ImageCollection('MODIS/006/MOD10A1')
+  .select('NDSI_Snow_Cover');
+```
+
+Mask pixels based on frequency of snow cover.
+
+```js
+// Pixels must have been 10% snow covered for at least 2 weeks in 2018.
+var snowCoverEphem = completeCol.filterDate('2018-01-01', '2019-01-01')
+  .map(function(img) {
+    return img.gte(10);
+  })
+  .sum()
+  .gte(14);
+
+// Pixels must not be 10% snow covered more than 124 days in 2018.
+var snowCoverConst = completeCol.filterDate('2018-01-01', '2019-01-01')
+  .map(function(img) {
+    return img.gte(10);
+  })
+  .sum()
+  .lt(125);
```

Comment:
Suggestion: Use `.lte(124)` to make this slightly easier to ready.

Reply:
Done